### PR TITLE
Disallow setting Parent on the started Activity

### LINKED ISSUE #81371: It should not be possible to set Activity parent after activity is started
### Description

If `Activity` has started without parent, and then `SetParentId` is called, then Activity.ParentId is updated.
If `Activity` had a parent at start time, `SetParentId` does nothing (correctly).

In distributed tracing, spans (activities) that are in parent/child relationship should have the same trace-id. Setting parent after activity is started (trace-id is generated, sampling decision is done) is invalid.

### Reproduction Steps

```csharp
using System.Diagnostics;

var a = new Activity("foo").Start();
Console.WriteLine($" trace id - {a.TraceId}, parent id - {a.ParentId}");
a.SetParentId("00-6e76af18746bae4eadc3581338bbe8b1-2899ebfdbdce904b-00");
Console.WriteLine($" trace id - {a.TraceId}, parent id - {a.ParentId}");
```

### Expected behavior

`SetParentId` on started `Activity` should be ignored (write first chance exception). 

### Actual behavior

```
 trace id - 0bb55e0092537a03fe5a83fc6030677e, parent id -
 trace id - 0bb55e0092537a03fe5a83fc6030677e, parent id - 00-6e76af18746bae4eadc3581338bbe8b1-2899ebfdbdce904b-00
```

### Regression?

not a regression, the issue was there from day 1

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes https://github.com/dotnet/runtime/issues/81371

### CODE REVIEW

**Comment by tarekgh** [comment]
@noahfalk @lmolkova

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs [hunk 1]
```diff
@@ -730,7 +739,15 @@ public void IdFormat_W3CInvalidVersionFF()
         [Fact]
         public void IdFormat_W3CWhenTraceIdAndSpanIdProvided()
         {
-            Activity activity = new Activity("activity3");
+            using (var a = new Activity("foo"))
```
Comment (noahfalk) [inline src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs:742]: Can we make a new test method for this code or move it to the other SetParentId test? The test method name sounds pretty accurate for the pre-existing code in this test but not for this new part.

**Review by noahfalk — APPROVED** [review]
LGTM

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs [hunk 2]
```diff
@@ -239,7 +239,16 @@ public void Tags()
...
         public void SetParentId()
         {
-            var parent = new Activity("parent");
+            using (var a = new Activity("foo"))
+            {
+                a.Start();
+                string parentId = a.ParentId;
+                a.SetParentId("00-6e76af18746bae4eadc3581338bbe8b1-2899ebfdbdce904b-00"); // Error does nothing
+                Assert.Equal(parentId, a.ParentId);
+
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs:248]: Nit:
```suggestion
```

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs:742]: I'll move it to the other test method.

**Pull request merged**